lib/util/db.js

```diff
@@ -241,6 +241,11 @@ const postgresErrorToProblem = (error) => {
         reason: 'not a recognized state name'
       }));
     }
+  } else if (error.code === '22001') {
+    const match = /value too long for type character varying\((\d+)\)/.exec(error.message);
+    if (match != null) {
+      return reject(Problem.user.valueTooLong({ reason: match[0] }));
```

Comment:
here you provide a reason, which is just the database error message. but in `lib/util/problem.js` you don't do anything with it, that parameter is thrown away. and, people using the web API won't know what a character varying is.

my suggestion is to make the detail object have `{ maxLength: # }`, and put the matched limit in there.

also, if there is any way (i didn't investigate deeply) to extract any kind of information about the actual input from the postgres error (eg the provided length) it would be nice to incorporate it here.

Reply:
Yes, I shall detail with `{maxLength : # }`
But,I don't think there is any way to know for which field he error occurred in this case. As the error message does not have sufficient information.
```
{ error: value too long for type character varying(64)
    at Connection.parseE (<path>/node_modules/pg/lib/connection.js:545:11)
    at Connection.parseMessage (<path>/node_modules/pg/lib/connection.js:370:19)
    at Socket.<anonymous> (<path>/node_modules/pg/lib/connection.js:113:22)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at TCP.onread (net.js:601:20)
  name: 'error',
  length: 98,
  severity: 'ERROR',
  code: '22001',
  detail: undefined,
  hint: undefined,
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'varchar.c',
  line: '624',
  routine: 'varchar' }
```

Reply:
does that `98` length property match with the attempted input? if it does, we can at least give that back to the user and that will be some hint as to which field went wrong.

Reply:
> oes that `98` length property match with the attempted input?

No `98` length property does not match with the attempted input

Reply:
that's too bad, okay